### ISSUE TITLE
Expense - Composer is missing for the invited member in the expense report

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#50933]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.49-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5088707
**Email or phone of affected tester (no customers):** applausetester+kh081006@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. [User A] Submit two expenses to User B.
3. [User A] Go to the expense report.
4. [User A] Send a message containing User C mention.
5. [User A] Invite User C by clicking Invite them on the whisper.
6. [User C] Open the invited expense report.

## Expected Result:
The composer should be present for the invited member (User C) in the expense report.

## Actual Result:
The composer is missing for the invited member (User C) in the expense report.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/27115b67-bb30-4209-88c6-f6c1e1cab957
 
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @francoisl</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by IuliiaHerets** [comment]
@puneetlath  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by huult** [comment]
I don’t think this is a bug. I checked, and User C does not have permission to send messages; this user, who was invited, can only read. I found this information in the logs.

- <img width="1512" alt="Screenshot 2024-10-17 at 14 46 35" src="https://github.com/user-attachments/assets/2c8f711a-8837-41dc-a009-84b9272f26d8">

**Comment by puneetlath** [comment]
Hm, interesting. Yeah I would think that we should give them the ability to comment. I'll look into that.

**Comment by puneetlath** [comment]
Haven't gotten to it yet.

**Comment by puneetlath** [comment]
Haven't had a chance to look into this one yet.

**Comment by puneetlath** [comment]
Going to un-assign, since this is a non-urgent #expense issue.

**Comment by kadiealexander** [comment]
Not overdue, not a priority.

**Comment by kadiealexander** [comment]
Still looking for volunteers

**Comment by kadiealexander** [comment]
Still not a priority, not overdue. Looking for anyone who wants to pick it up!

**Comment by kadiealexander** [comment]
Not overdue, requesting a retest.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by francoisl** [comment]
Moving to weekly, will close next week if it's confirmed fixed after another retest
